### PR TITLE
fix memset for route_hop json parsing when no channel_id exists.

### PR DESCRIPTION
**Pull request by fiatjaf**

### CODE REVIEW

File: lightningd/pay.c [hunk 1]
```diff
@@ -1163,7 +1163,7 @@ param_route_hop(struct command *cmd, const char *name, const char *buffer,
 	}
 
 	if (!channeltok) {
-		memset(&res->channel_id, 0, sizeof(struct node_id));
+		memset(&res->channel_id, 0, sizeof(struct short_channel_id));
```
Comment (darosior) [inline lightningd/pay.c:1166]: Nice catch

**Review by darosior — APPROVED** [review]
ACK c60db2d1bca715bdb6906b6c00b39645c4c6ca38

**Comment by darosior** [comment]
Can you add a `Changelog-None` ?

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/4306/commits/24eabb898f701b09672797b073c6706d9dee09bf

**Pull request merged**